### PR TITLE
feat(TTD): Implement the due dates of a task

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
Description: 
- Implemented the functionality for users to add sue dates to a task.

Associated Ticket Link(s):
- https://trello.com/c/0TdnHChA/12-add-due-dates

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ function App() {
           task: todo,
           isCompleted: false,
           priority: Priority.HIGH,
-          deadline: new Date(),
+          deadline: { from: new Date(), to: undefined },
```
Comment (mMohsinSazim) [inline src/App.tsx:31]: without explicitly saying undefined, make that property optional

File: src/Model.ts [hunk 2]
```diff
@@ -8,5 +8,5 @@ export interface Tasks{
     task: string,
     isCompleted: boolean,
     priority?: Priority,
-    deadline: Date;
+    deadline?: { from: Date | undefined; to: Date | undefined };
```
Comment (mMohsinSazim) [inline src/Model.ts:11]: without explicitly saying undefined, make that property optional

File: src/components/SingleTask.tsx [hunk 3]
```diff
@@ -41,10 +40,7 @@ interface Props {
 }
 
 const SingleTask = ({ task, tasks, setTasks }: Props) => {
-  const [date, setDate] = React.useState<DateRange | undefined>({
-    from: new Date(2022, 0, 20),
-    to: addDays(new Date(2022, 0, 20), 20),
-  });
+  const [date, setDate] = React.useState<DateRange | undefined>(task.deadline);
```
Comment (mMohsinSazim) [inline src/components/SingleTask.tsx:43]: use default export from React instead of React.useState

**Review by mMohsinSazim — CHANGES_REQUESTED** [review]

File: src/Model.ts (hunk 2 above)
Comment (mohammed-sharraf-sazim) [reply src/Model.ts]: Resolved the issue

File: src/components/SingleTask.tsx (hunk 3 above)
Comment (mohammed-sharraf-sazim) [reply src/components/SingleTask.tsx]: Resolved the issue

**Review by mMohsinSazim — APPROVED** [review]

**Pull request merged**